File: mifosng-android/src/main/java/com/mifos/objects/client/ClientAddressResponse.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.objects.client;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class ClientAddressResponse implements Parcelable {

    @SerializedName("id")
    Integer id;

    @SerializedName("client_id")
    Integer clientId;

    @SerializedName("latitude")
    Double latitude;

    @SerializedName("longitude")
    Double longitude;

    @SerializedName("place_address")
    String placeAddress;

    @SerializedName("place_id")
    String placeId;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getClientId() {
        return clientId;
    }

    public void setClientId(Integer clientId) {
        this.clientId = clientId;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public String getPlaceAddress() {
        return placeAddress;
    }

    public void setPlaceAddress(String placeAddress) {
        this.placeAddress = placeAddress;
    }

    public String getPlaceId() {
        return placeId;
    }

    public void setPlaceId(String placeId) {
        this.placeId = placeId;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(this.id);
        dest.writeValue(this.clientId);
        dest.writeValue(this.latitude);
        dest.writeValue(this.longitude);
        dest.writeString(this.placeAddress);
        dest.writeString(this.placeId);
    }

    public ClientAddressResponse() {
    }

    protected ClientAddressResponse(Parcel in) {
        this.id = (Integer) in.readValue(Integer.class.getClassLoader());
        this.clientId = (Integer) in.readValue(Integer.class.getClassLoader());
        this.latitude = (Double) in.readValue(Double.class.getClassLoader());
        this.longitude = (Double) in.readValue(Double.class.getClassLoader());
        this.placeAddress = in.readString();
        this.placeId = in.readString();
    }

    public static final Parcelable.Creator<ClientAddressResponse> CREATOR =
            new Parcelable.Creator<ClientAddressResponse>() {
        @Override
        public ClientAddressResponse createFromParcel(Parcel source) {
            return new ClientAddressResponse(source);
        }

        @Override
        public ClientAddressResponse[] newArray(int size) {
            return new ClientAddressResponse[size];
        }
    };
}

File: core/model/src/commonMain/kotlin/com/mifos/core/model/objects/clients/ClientAddressResponse.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.model.objects.clients

import com.mifos.core.model.utils.Parcelable
import com.mifos.core.model.utils.Parcelize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
@Parcelize
data class ClientAddressResponse(
    var id: Int? = null,

    @SerialName("client_id")
    var clientId: Int? = null,

    var latitude: Double? = null,

    var longitude: Double? = null,

    @SerialName("description")
    var placeAddress: String? = null,

    var placeId: String? = null,
) : Parcelable
